lib/protocol/websocket/text_frame.rb

```diff
@@ -29,12 +29,21 @@ def data?
 				true
 			end
 			
-			def unpack
-				super.force_encoding(Encoding::UTF_8)
-			end
+			# NOTE: This is here as a reminder that there is no guarantee that a
+			#       frame is valid UTF-8, only the full message.
+			#
+			# FIXME: In theory we should fail fast on invalid codepoints here.
+			#
+			# def unpack
+			#   super
+			# end
 			
 			def pack(data)
-				super(data.b)
```

Comment:
@rawrafox do you mind explaining the logic here? My understanding is that we should be packing binary data, not UTF-8.

Reply:
Or is should we have `super(data.encode(UTF_8).b)`?

Reply:
The code "after" this converts it to a binary anyhow, so there is no point in doing the work twice.

For masked payloads this is done in `Frame#pack` and for non-masked payloads this is done in `Frame#write` when written to the socket.

Reply:
Okay so the point of encoding here is to change the encoding to UTF-8 if needed before converting to binary. That makes sense.